### ISSUE TITLE
Recent build failures

### ISSUE THREAD

**Issue by tdammers** [input-output-hk/cardano-base#321]
As of this week, I am experiencing build failures of `cardano-base` and some other repos, despite the same code building just fine earlier.

An illustrative example can be found in these CI workflow runs:

- https://github.com/input-output-hk/cardano-base/actions/runs/3252202593 completed without errors last week.
- https://github.com/input-output-hk/cardano-base/actions/runs/3280522058 failed today.

Both these runs are based on the exact same code; I simply made a new branch off of `master`, and pushed it without any further changes.

The relevant part from the error log is this:

```
cabal: Could not resolve dependencies:
[__0] trying: cardano-binary-1.5.0.1 (user goal)
[__1] rejecting: cardano-binary:!test (constraint from config file, command
line flag, or user target requires opposite flag selection)
[__1] trying: cardano-binary:*test
[__2] unknown package: cardano-prelude-test (dependency of cardano-binary
*test)
[__2] fail (backjumping, conflict set: cardano-binary, cardano-prelude-test,
cardano-binary:test)
After searching the rest of the dependency tree exhaustively, these were the
goals I've had most trouble fulfilling: cardano-binary, cardano-binary:test,
cardano-prelude-test
```

Which suggests that for some reason, `cardano-prelude-test` cannot be found anymore, despite `cabal.project` still referring to the same commit hashes as before, which should in theory guarantee reproducible builds.

The same failure also occurs on my local machine.

**Comment by tdammers** [comment]
Tagging @lehins

**Comment by iquerejeta** [comment]
Resolved by #322.
